src/LogFileReader.cpp

```diff
@@ -47,12 +49,44 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
+    unsigned int allowedCharLength = 15;
+    std::string allowedChars [allowedCharLength] = {"1","2","3","4","5","6","7","8","9","0",":",",",".","-"," "};
```

Comment:
When you can, you should try to use the built in types, in this case ` QString` would fit better instead of `std::string`.